src/libraries/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -392,7 +392,29 @@ public static void SetWindowPosition(int left, int top)
 
         public static void SetWindowSize(int width, int height)
         {
-            throw new PlatformNotSupportedException();
+           lock (Console.Out)
+           {
+               Interop.Sys.WinSize winsize = default;
+               winsize.Row = (ushort)height;
+               winsize.Col = (ushort)width;
+               if (Interop.Sys.SetWindowSize(in winsize) == 0)
+               {
+                   s_windowWidth = winsize.Col;
+                   s_windowHeight = winsize.Row;
+               }
+               else
+               {
+                   Interop.ErrorInfo errorInfo = Interop.Sys.GetLastErrorInfo();
+                   if (errorInfo.Error == Interop.Error.ENOTSUP)
+                   {
+                       throw new PlatformNotSupportedException();
+                   }
+                   else
+                   {
+                       throw Interop.GetIOException(errorInfo);
+                   }
```

Comment:
Nit: this could be consolidated a bit to:
```suggestion
                   throw errorInfo.Error == Interop.Error.ENOTSUP ?
                       new PlatformNotSupportedException() :
                       Interop.GetIOException(errorInfo);
```